HealthyPlanners/Urban-Sherpa
# Implement loader

**Implemented Loader Icons for the Three APIs currently in use**
Created functionality to change state and render Wheelloader based on state in Long Lat Test component

**Should...**
Render a loading wheel when api calls are made for walk, yelp, and IQAir. Loaders should disappear when promises is either resolved or rejected for said api calls.

**To Test**
-Sign in, choose an address, and click "Get Health". Loaders should briefly render and then be replaced by results of API calls.
-Further testing - Close server, refresh site, and repeat above steps. Loaders should VERY BRIEFLY render and then disappear when api calls fail.
-Furthest Testing - To see extended loaders, venture to client/store/entities/mapEntities and comment out lines 96 - 99. Save, refresh the app and follow steps at the top of this section (server closed and make an address selection). 
![image](https://user-images.githubusercontent.com/55901780/85933126-458aaa80-b888-11ea-981d-043baf538b35.png)

Loaders should persist indefinitely.

**Files Touched**
**client/store/entities/mapEntity**
-Restructured data for Clean, api specific loading flag
-Added functionality to differentiate which api calls are triggering data request and data failed actions

**client/store/middleware/apiCall**
-Cleaned up some errors
-Enabled passing of startPayload and errorPayload to differentiate which API call is triggering data request and failed actions

**client/components/experiemental/newerMapComponent**
-Accounted for new data structure in state destructuring

**client/components/experimental/LongLatTest**
-Accounted for new data structure in props destructuring -- Should allow for minimal impact to anyone else working off of data in this component
-Added functionality for conditional render on loaders

NOTE - All other files in this PR are captured and described in another PR as part of a different branch.
